### ISSUE TITLE
Australia OPENNEM exchange parser broke

### ISSUE THREAD

**Issue by jmdesp** [electricitymap/electricitymap-contrib#3689]
The date for South Australia is showing exactly the same emission intensity for consumption and generation, despite the fact that import can be at time a significant part of the mix.
It reports using data from opennem that seems however to allow capturing that information.
On this page an export button to csv is available https://opennem.org.au/energy/sa1/?range=1d&interval=30m and the csv does contain import and export columns.
They are some other pages here with equivalent data but maybe somewhat different : http://nemlog.com.au/gen/region/sa/
While in some case the data is shown as png, the interconnector page http://nemlog.com.au/interconnector/ has a csv link and maybe more precise data than opennem if each interconnector can be seen separately.
http://nemlog.com.au/api/ic/NSW1-QLD1/csv The co2 page has a csv link too http://nemlog.com.au/gen/co2e/

**Comment by jmdesp** [comment]
A capture of the opennem display showing a significant participation of imports for the other more emitting regions of Australia that isn't reflected on electricity map
![image](https://user-images.githubusercontent.com/842054/150206049-98eaf6c6-0fde-4ea9-8547-e455f2ab80f7.png)
What is reported by electricity map for the same period :
![image](https://user-images.githubusercontent.com/842054/150206776-c8801c96-2985-4751-89c0-891102db1b94.png)

**Comment by jarek** [comment]
Hi, thanks for reporting!

Looks like the parsers for exchanges are failing, e.g.:

https://storage.googleapis.com/electricitymap-parser-logs/AUS-SA.html

```
KeyError	'region'

  File "/home/contrib/parsers/OPENNEM.py", line 384, in fetch_exchange
    logger=logger,

  File "/home/contrib/parsers/OPENNEM.py", line 206, in fetch_main_power_df
    capacities = get_capacities(filtered_datasets, region)

  File "/home/contrib/parsers/OPENNEM.py", line 104, in get_capacities
    for obj in filtered_datasets

  File "/home/contrib/parsers/OPENNEM.py", line 105, in <listcomp>
    if obj["region"] == region          
```

and the same in https://storage.googleapis.com/electricitymap-parser-logs/AUS-NSW.html for AUS-NSW->AUS-VIC and AUS-NSW->AUS-QLD

**Comment by skovhus** [comment]
@jarek thanks for investigating. I'm happy that you were able to discover this in the new parser logs. 👏

**Comment by jmdesp** [comment]
Does this need a help wanted tag ?

**Comment by chris-elia** [comment]
Hello everyone

Some colleagues and I had a look at the issue today. This is the first issue I am working on and I don't have a solution yet. I will explain a bit more in detail where the problem occurs and one possible solution idea. I would be happy for your feedback and ideas. Below you find a screenshot summarizing the functions and the important parameters that are called. I explain it in the text below.

![grafik](https://user-images.githubusercontent.com/95425912/156595015-649bd4d1-39b2-4a56-ba0a-64027f5d5c33.png)

The problem occurs as described above in the _fetch_exchange_ function. I tested it with the regions "AUS-SA" and "AUS-VIC".
The _fetch_exchange_ function is called with the two zone_keys ("AUS-SA","AUS-VIC") and the variable sorted_zone_keys is created. Next, the function _fetch_main_power_df_ is called. It is possible to pass a zone_key to this function but is left blanc. Only sorted_zone_keys is passed. 
Within _fetch_main_power_df_, zone_key is set to None. Then __fetch_man_df_ is called. As zone_key is None, region is also None, which leads to the problem in the function _get_capacities_  that region cannot be found. 
In addition, in the filtered_dataset the key "region" does not exist.(See screenshot). In my point of view the keys "id" or "code" could be helpful to determine the flow direction.

![grafik](https://user-images.githubusercontent.com/95425912/156594807-5fc9e506-a310-43de-9562-d75e1bcf0059.png)

The _get_capacity_ function is used in _fetch_exchange_ and _fetch_production_. The url call is different when _fetch_exchange_ and _fetch_production_ are called. I assume that the results are also different but surprisingely _get_capacity_ function is called for both functions without differentiating the data.

I dont know how it was originally working. A possible idea on how to solve the issue could be to create a custom get_capacity function for the fetch_exchange function. 

Do you think that this is going in the right direction? Does anybody know the original behavior of the function?

Kind regards
Chris.

**Comment by madsnedergaard** [comment]
Hey @chris-elia, thanks for the thorough investigation! I just tested this again, and it seems like it worked before https://github.com/electricitymap/electricitymap-contrib/pull/3550 - so something in that PR broke exchanges :)

**Comment by madsnedergaard** [comment]
And you're right, we shouldn't try to get capacities when we're fetching exchanges! The PR mentioned above changed how fetch_main_df works, and it seems like we the different values it returns.

In short, the following change fixes the problem:
<img width="984" alt="Screenshot 2022-03-03 at 20 32 11" src="https://user-images.githubusercontent.com/3296643/156639707-b7c5d5a7-f7c7-4f96-bfd1-11a0f4b08dd2.png">

**Comment by chris-elia** [comment]
Hello  @madsnedergaard 

thank you very much for your feedback.! Some colleagues and I continued to work on the issue today and tested your solution. Unfortunately, we did not achieve to test it because an assertion error was caused before. ( function data_to_df -> assert statement where len(index) is compared with len(series["data"])
During the investigation it looked like that the data that is returned from the API does not include all values for every 5 minute step. We would expect 288/289 values (24h*60min/5min) but only got 284. I think this is another issue but maybe already known.
I assume that the problem might not be there anymore tomorrow. I will check next week again if it works.
Thanks for your quick feedback and help.
Kind regards
Chris. 

![image](https://user-images.githubusercontent.com/95425912/156842589-d26489af-1d6a-4afb-8913-bb6ff266fb7d.png)
